SConstruct

```diff
@@ -205,7 +196,67 @@ if not ('CC' in main and 'CXX' in main):
     error("No C++ compiler installed (package g++ on Ubuntu and RedHat)")
 
 # Find default configuration & binary.
-Default(environ.get('M5_DEFAULT_BINARY', 'build/ARM/gem5.debug'))
+default_target = environ.get('M5_DEFAULT_BINARY', None)
+if default_target:
+    Default(default_target)
+
+# If no target is set, even a default, print help instead.
+if not BUILD_TARGETS:
+    warning("No target specified, and no default.")
+    SetOption('help', True)
+
+buildopts_dir = Dir('#build_opts')
+buildopts = list([f for f in os.listdir(buildopts_dir.abspath) if
+        isfile(os.path.join(buildopts_dir.abspath, f))])
+buildopts.sort()
+
+buildopt_list = '\n'.join(' ' * 10 + buildopt for buildopt in buildopts)
+
+Help(f"""
+Targets:
+        To build gem5 using a predefined configuration, use a target with
+        a directory called "build" in the path, followed by a directory named
+        after a predefined configuration, and then the actual target, likely
+        a gem5 binary. For example:
+
+        scons build/X86/gem5.opt
+
+        The "build" component tells SCons that the next part names an initial
+        configuration, and the part after that is the actual target.
+        The predefined targets currently available are:
+
+{buildopt_list}
+
+        The extension on the gem5 binary specifies what type of binary to
+        build. Options are:
+
+        debug: A debug binary with optimizations turned off and debug info
+            turned on.
+        opt: An optimized binary with debugging still turned on.
+        fast: An optimized binary with debugging, asserts, and tracing
+            disabled.
+
+        gem5 can also be built as a static or dynamic library. In that case,
+        the extension is fixed by the operating system, so the binary type
```

Comment:
> the extension is fixed by the operating system

What does this mean? Do we support any OSes that use an extension other than `.so`?

Reply:
Yes, the users should determine the extension name beforehand to build gem5 dynamic library. For example, `.so` for linux platform and `.dylib` for MacOS. I have updated the help description to include the build commands for Linux and MacOS respectively. The users should run the corresponding build command in their OS.